File: src/titles.rs
pub struct TitleSystem;

impl TitleSystem {
    /// 锁死为恰好 5 个中文字符（在 Ratatui 终端中占用 10 显示列）
    pub fn get_title_by_level(level: u32) -> &'static str {
        match level {
            1..=5    => "凡铁打刀客",
            6..=10   => "赛博小学徒",
            11..=20  => "低空炼器师",
            21..=30  => "五行锻造客",
            31..=45  => "赛博宗大师",
            46..=60  => "虚空淬火尊",
            61..=80  => "天道炉火仙",
            81..=99  => "熵增造物神",
            _        => "极客大道祖",
        }
    }
}

File: server/titles.ts
// server/titles.ts

export class TitleSystem {
  public static getTitleByLevel(level: number): string {
    if (level <= 15) return '凡铁打刀客';
    if (level <= 30) return '赛博引气匠';
    if (level <= 45) return '纳米识海者';
    if (level <= 60) return '高能内核体';
    if (level <= 75) return '赛博数字婴';
    if (level <= 90) return '量子态化神';
    if (level <= 105) return '矩阵合体尊';
    if (level <= 120) return '虚空淬火尊';
    if (level <= 135) return '反重力真仙';
    if (level <= 150) return '因果律圣者';
    if (level <= 165) return '天道炉火仙';
    return '熵增造物神';
  }
}
